packages/connection-storage/src/connection-storage.ts

```diff
@@ -98,11 +107,128 @@ export class ConnectionStorage {
     this.calledOnce = true;
   }
 
-  private static mapStoredConnectionToConnectionInfo(
-    storedConnectionInfo: ConnectionInfo,
+  static async migrateToSafeStorage() {
+    // If user lands on this version of Compass with legacy connections (using storage mixin),
+    // we will ignore those and only migrate connections that have connectionInfo.
+    const connections = (await this.getConnections()).filter(
+      (x) => x.connectionInfo
```

Comment:
very optional, but makes it clear below that the filtering has already actually happened:

```suggestion
      (x): x is ConnectionWithLegacyProps & Required<Pick<ConnectionProps, 'connectionInfo'>> => !!x.connectionInfo
```